components/formats-gpl/src/loci/formats/in/BDVReader.java

```diff
@@ -0,0 +1,872 @@
+          String path_to_site = path_to_setup + "/" + resolution;
+          LOGGER.info("Site :" + path_to_site );  
+          H5PositionList.add(new H5Coordinate(String.format("t%05d", timepoint), setup, resolution));
+        }
+      }
+    }
+
+    if (H5PositionList.size() == 0) {
+      throw new FormatException("No series found in file...");
+    }
+
+    List<String> seriesNames = new ArrayList<String>();
+
+    String formattedFirstTimepoint = String.format("t%05d", firstTimepoint);
+    if (sizeC == 0) sizeC = 1;
+    int sumOfResolutions = 0;
+    for (H5Coordinate coord : H5PositionList) {
+      if (jhdf.exists(coord.pathToImageData)) {
+        H5PathsToImageData.add(coord.pathToImageData);
+        
+        // Don't create new series for each timepoint
+        if (coord.timepoint.equals(formattedFirstTimepoint)) {
+          int setupIndex = Integer.parseInt(coord.setup.substring(1));
+          HashMap<String, String> setupAttributes = setupAttributeList.get(setupIndex);
+          String firstChannelIndex = channelIndexes.size() > 0 ? channelIndexes.get(0).toString() : "";
+          
+          // Dont create a new series for each channel
+          if (sizeC == 1 || (setupAttributes.containsKey("channel") && setupAttributes.get("channel").equals(firstChannelIndex))) {
+            CoreMetadata m = new CoreMetadata();
+            core.add(m);
+            int resolutionsInThisSetup = setupResolutionCounts.get(setupIndex);
+            if (hasFlattenedResolutions()) {
+              setSeries(seriesCount);
+            }
+            else {
+              setSeries(coreIndexToSeries(seriesCount));
+              setResolution(seriesCount - sumOfResolutions);
+              if (seriesCount == sumOfResolutions + resolutionsInThisSetup - 1) {
+                sumOfResolutions += resolutionsInThisSetup;
+              }
+            }
+    
+            LOGGER.debug(coord.pathToImageData);
+            int[] ctzyx = jhdf.getShape(coord.pathToImageData);
+            m.sizeC = sizeC;
+            m.sizeT = numTimepoints;
+            m.sizeZ = ctzyx[0];
+            m.sizeY = ctzyx[1];
+            m.sizeX = ctzyx[2];
+            m.resolutionCount = resolutionsInThisSetup;
+            m.thumbnail = false;
+            m.imageCount = getSizeC() * getSizeT() * getSizeZ();
+            m.dimensionOrder = "XYZTC";
+            m.rgb = false;
+            m.thumbSizeX = 128;
+            m.thumbSizeY = 128;
+            m.orderCertain = false;
+            m.littleEndian = true;
+            m.interleaved = false;
+            m.indexed = true;
+            int bpp = jhdf.getElementSize(coord.pathToImageData);
+            if (bpp==1) {
+              m.pixelType = FormatTools.UINT8;
+            }
+            else if (bpp==2) {
+              m.pixelType = FormatTools.UINT16;
+            }
+            else if (bpp==4) {
+              m.pixelType = FormatTools.INT32;
```

Comment:
Interesting it's not also UINT.

Reply:
From the spec that I have the data coming from Big Data Viewer should be UINT16:

```
BigDataViewer always stores images with 16-bit precision currently, while the image you want to export is not necessarily 16-bit. The value range defines the minimum and maximum of the image you want to export. This is mapped to the 16-bit range for export. I.e., the minimum of the value range will be mapped to the minimum of the unsigned 16-bit range (0). The maximum of the value range will be mapped to the maximum of the unsigned 16-bit range (65535).
```

The other options (namely the out of place INT32) are inline with what we have for other H5 files such as CellH5 which has UINT16 and INT32

Reply:
Thanks for opening the discussion. The samples we have are all 16-bit as per the specification that David linked from. I think keeping this handling should be largely fine and we can review this if the specification get extended with representative samples. Alternatively, we could be conservative and decide to throw an exception if `bpp` is 4.